ci/run_clang_tidy.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# Copyright (c) 2020-2023, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+rapids-logger "Create clang_tidy conda environment"
+. /opt/conda/etc/profile.d/conda.sh
+
+rapids-dependency-file-generator \
+  --output conda \
+  --file_key clang_tidy \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee env.yaml
+
+rapids-mamba-retry env create --force -f env.yaml -n clang_tidy
+# Temporarily allow unbound variables for conda activation.
+set +u && conda activate clang_tidy && set -u
+
+./build.sh --nobuild libcuml
+
+rapids-logger "Run clang-tidy"
+
+python cpp/scripts/run-clang-tidy.py -ignore '[.]cu$|_deps|examples/kmeans/'
```

Comment:
Can we get a section in https://github.com/rapidsai/cuml/blob/branch-23.08/CONTRIBUTING.md#code-formatting that quickly describes how to run this locally for devs?

Reply:
Addressed in https://github.com/rapidsai/cuml/pull/5235/commits/c4e656531d62b497f9e758f0dcc2839019ece1f4 .